src/prototype.rs

```diff
@@ -144,7 +144,38 @@ impl<'a> FirstPass<'a> {
             return self.parse_indented_code_block(ix, remaining_space);
         }
 
+
+        // HTML Blocks
+
+        // HTML blocks may have 1-3 spaces in front of them, but our scanning
+        // functions don't know that. The spaces are preserved in the output,
+        // so we don't want to move `ix` yet.
         line_start.scan_all_space();
```

Comment:
I think the `scan_all_space` here is a no-op, any remaining space should have been consumed by the `scan_space_upto(4)` above. Maybe try removing it and verify nothing breaks?